### ISSUE TITLE
Make `select` and `oc.select` more robust

### ISSUE THREAD

**Issue by SZiesche** [omry/omegaconf#1127]
**Is your feature request related to a problem? Please describe.**
By using `oc.select` we can define default values if the interpolation target is not found. However, the "not found" part is not very robust. For example, if we refer to a non-existing node, we still raise an error.

Consider for example the case, where we have a config, that is usually nested in some other config and has a member like `my_member: int = ${oc.select:..member, 5}` that is by default inherited from the parent. Now we want to test this config standalone or we want to use it in a different context were the parent doesn't exist, then the code just crashes. 

**Describe the solution you'd like**
It would be cool to turn this off or to give the user an option to turn this off. For example, we could change the select implementation to 

```python
def select_value(
    cfg: Container,
    key: str,
    *,
    default: Any = _DEFAULT_MARKER_,
    throw_on_resolution_failure: bool = True,
    throw_on_missing: bool = False,
    absolute_key: bool = False,
) -> Any:
    try:  # this try/except block is new, the rest stays the same.
        node = select_node(
            cfg=cfg,
            key=key,
            throw_on_resolution_failure=throw_on_resolution_failure,
            throw_on_missing=throw_on_missing,
            absolute_key=absolute_key,
        )
    except ConfigKeyError as e:
        node = None

    node_not_found = node is None
    if node_not_found or node._is_missing():
        if default is not _DEFAULT_MARKER_:
            return default
        else:
            return None

    return _get_value(node)
```

**Describe alternatives you've considered**
Alternatively, we could add a flag like `throw_on_node_not_existing` to the function and raise the error depending on that value. Then the users could relatively easy write their own robust select method.

**Comment by odelalleau** [comment]
Indeed, it shouldn't crash if it can't find the node when a default value is provided.

**Comment by SZiesche** [comment]
to make this even more precise, I would like to have the following script pass.
```python
from dataclasses import dataclass

from omegaconf import OmegaConf, II


cfg1 = OmegaConf.create({"a": "${oc.select:..member, 5}"})
print(cfg1.a)  # this raises an Error because of the missing node


@dataclass
class Config:
    a: int = II("${oc.select:..member, 5}")


cfg2 = OmegaConf.structured(Config)
print(cfg2.a)  # this raises an Error too
```

**Comment by omry** [comment]
A note on the structured-config example above:

```python
@dataclass
class Config:
    a: int = II("${oc.select:..member, 5}")
```

This example is not equivalent to assigning `a` the value returned by `oc.select`.

`II(x)` wraps its argument in `${...}`, so this becomes:

```yaml
a: ${${oc.select:..member, 5}}
```

That is a dynamic interpolation. First OmegaConf evaluates the inner interpolation, then it uses the result as the key for the outer interpolation.

So if the inner select returns `some_key`, OmegaConf resolves:

```yaml
${some_key}
```

But if the inner select falls back to `5`, OmegaConf tries to resolve:

```yaml
${5}
```

That is not the same as returning the value `5`; it means "interpolate the node identified by `5`". In this context the default must have the same semantics as the value returned by `oc.select`: it must be a valid interpolation target.

For example, this works because the default points to an existing node:

```yaml
fallback: 123
ok: ${${oc.select:..member, fallback}}
```

and this correctly raises because the default points nowhere:

```yaml
bad: ${${oc.select:..member, nowhere}}
```

If the intent is to use `5` as the actual fallback value, the structured config should use a string interpolation directly, for example:

```python
@dataclass
class Config:
    a: int = SI("${oc.select:..member, 5}")
```

or the equivalent raw interpolation string. The plain `oc.select` behavior itself is still valid to fix: when the relative parent path does not exist and a default is provided, it should return the default rather than crashing.
